apps/fortuna/src/config.rs

```diff
@@ -155,65 +134,86 @@ pub struct EthereumConfig {
-            anyhow!(
-                "Could not find chain id {} in provider configuration",
-                &chain_id
-            )
-            .into(),
-        )
-    }
+    /// The provider's private key, which is required to register, update the commitment,
+    /// or claim fees. This argument *will not* be loaded for commands that do not need
+    /// the private key (e.g., running the server).
+    pub private_key: SecretString,
+
+    /// The provider's secret which is a 64-char hex string.
+    /// The secret is used for generating new hash chains
+    pub secret: SecretString,
+
+    /// The length of the hash chain to generate.
+    pub chain_length: u64,
 }
 
+/// Configuration values for the keeper service that are shared across chains.
 #[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
-pub struct ProviderChainConfig {
-    commitments: Option<Vec<Commitment>>,
-    pub fee:     u128,
+pub struct KeeperConfig {
+    /// If provided, the keeper will run alongside the Fortuna API service.
+    /// The private key is a 20-byte (40 char) hex encoded Ethereum private key.
+    /// This key is required to submit transactions for entropy callback requests.
+    /// This key *does not need to be a registered provider*. In particular, production deployments
+    /// should ensure this is a different key in order to reduce the severity of security breaches.
+    pub private_key: SecretString,
 }
 
-impl ProviderChainConfig {
-    /// Returns a clone of the commitments in the sorted order.
-    /// `HashChainState`  requires offsets to be in order.
-    pub fn get_sorted_commitments(&self) -> Vec<Commitment> {
-        let mut commitments = self.commitments.clone().unwrap_or(Vec::new());
-        commitments.sort_by(|c1, c2| {
-            c1.original_commitment_sequence_number
-                .cmp(&c2.original_commitment_sequence_number)
-        });
-        commitments
-    }
+// A secret is a string that can be provided either as a literal in the config,
+// or in a separate file. (The separate file option is useful for 1password mounting in production.)
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
+pub struct SecretString {
+    pub value: Option<String>,
+
+    // The name of a file containing the string to read. Note that the file contents is trimmed
+    // of leading/trailing whitespace when read.
+    pub file: Option<String>,
 }
 
-#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
-pub struct Commitment {
-    pub seed:                                [u8; 32],
-    pub chain_length:                        u64,
-    pub original_commitment_sequence_number: u64,
+impl SecretString {
+    pub fn load(&self) -> Result<Option<String>> {
+        match &self.value {
+            Some(v) => return Ok(Some(v.clone())),
+            _ => {}
+        }
+
+        match &self.file {
+            Some(v) => {
+                return Ok(Some(fs::read_to_string(v)?.trim().to_string()));
```

Comment:
the trim here is possibly controversial but i lost like 20 mins on this because my hex file had a newline at the end

Reply:
it's not. I also spent some time yesterday fixing this in another place and ended up adding a trim there too.